gateway/src/main/java/io/zeebe/gateway/grpc/GrpcErrorMapper.java

```diff
@@ -99,6 +99,20 @@ private Status mapBrokerErrorToStatus(final BrokerError error) {
       case RESOURCE_EXHAUSTED:
         builder.setCode(Code.RESOURCE_EXHAUSTED_VALUE);
         break;
+      case PARTITION_LEADER_MISMATCH:
+        // return UNAVAILABLE to indicate to the user that retrying might solve the issue, as this
+        // is usually a transient issue
+        builder.setCode(Code.UNAVAILABLE_VALUE);
+        break;
+        // all the following are not errors which retrying (with the same gateway) will solve
+      case INVALID_MESSAGE_TEMPLATE:
+      case INVALID_DEPLOYMENT_PARTITION:
+      case MALFORMED_REQUEST:
+      case INVALID_CLIENT_VERSION:
+      case UNSUPPORTED_MESSAGE:
+      case INTERNAL_ERROR:
+      case SBE_UNKNOWN:
+      case NULL_VAL:
       default:
         builder.setCode(Code.INTERNAL_VALUE);
         message =
```

Comment:
Maybe it makes more sense to return the string representation of the error code,  instead of returning the code in the error message? Then it is more clear for the user/client?

Reply:
Probably, - I can't remember now why we return the code and not the string representation.

Reply:
By string representation, you mean of the code (in this case that's what we do), or you mean the error message? The idea why we don't put the error message is that for these errors, we've logged it on the broker side, and it's probably not very useful for the client (also, it may contain sensitive info for some customers such as file path, system info, etc.)

Reply:
Ok my fault sorry. I thought this returns the int representation.